MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/MainActivity.java

```diff
@@ -1,6 +1,61 @@
 package com.google.android.libraries.motionphotoreader;
 
 import android.app.Activity;
+import android.os.Build;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+
+import androidx.annotation.RequiresApi;
+
+import com.adobe.internal.xmp.XMPException;
+
+import java.io.IOException;
 
 public class MainActivity extends Activity {
+
+    private static final String TAG = "MainActivity";
+    private static final String[] FILENAMES = { // replace these with appropriate file access
+            "20200621_200240",
+            "20200616_124008",
+            "20200621_184700"
+    };
+
+    private String filename;
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        filename = "/sdcard/MVIMG_" + FILENAMES[0] + ".jpg";
+        MotionPhotoWidget motionPhotoWidget = findViewById(R.id.motion_photo_widget);
+        Log.d(TAG, "Motion photo widget set up");
+        try {
+            motionPhotoWidget.setFile(filename);
+        } catch (IOException | XMPException e) {
+            e.printStackTrace();
+        }
+        Log.d(TAG, "Set file to " + filename);
+        motionPhotoWidget.setOnClickListener(new MyOnWidgetClickListener(motionPhotoWidget));
+    }
+
+    private class MyOnWidgetClickListener implements View.OnClickListener {
```

Comment:
fix naming to something more appropriate, e.g. `WidgetClickListener`

Reply:
Went with lambda function (suggestion from last time).